core/domain/user_jobs_continuous.py

```diff
@@ -293,100 +291,72 @@ def _handle_incoming_event(cls, active_realtime_layer, event_type, *args):
     @classmethod
     def _get_continuous_computation_class(cls):
         return UserImpactAggregator
 
-    MULTIPLIER = 10
-    MIN_AVERAGE_RATING = 2.5
-    NUM_RATINGS_SCALER_CUTOFF = 10
-    NUM_RATINGS_SCALER = .1
-
     @classmethod
     def entity_classes_to_map_over(cls):
-        return [exp_models.ExplorationModel]
-
-    @classmethod
-    def _get_exp_impact_score(cls, exploration_id):
-        # Get ratings and compute average rating score.
-        rating_frequencies = (
-            rating_services.get_overall_ratings_for_exploration(
-                exploration_id))
-        total_num_ratings = 0
-        total_rating = 0.0
-        for rating, num_ratings in rating_frequencies.iteritems():
-            total_rating += (int(rating) * num_ratings)
-            total_num_ratings += num_ratings
-        # Only divide by a non-zero number.
-        if total_num_ratings == 0:
-            return 0
-        average_rating = total_rating / total_num_ratings
-
-        # Get rating term to use in impact calculation.
-        rating_term = average_rating - cls.MIN_AVERAGE_RATING
-        # Only explorations with an average rating greater than the minimum
-        # have an impact.
-        if rating_term <= 0:
-            return 0
-
-        # Get num_ratings_scaler.
-        if total_num_ratings < cls.NUM_RATINGS_SCALER_CUTOFF:
-            num_ratings_scaler = cls.NUM_RATINGS_SCALER * total_num_ratings
-        else:
-            num_ratings_scaler = 1.0
-
-        # Get number of completions/playthroughs.
-        num_completions = (
-            stats_jobs_continuous.StatisticsAggregator.get_statistics(
-                exploration_id, stats_jobs_continuous.VERSION_ALL)[
-                    'complete_exploration_count'])
-        # Only take the log of a non-zero number.
-        if num_completions <= 0:
-            return 0
-        num_completions_term = math.log(num_completions)
-
-        exploration_impact_score = (
-            rating_term *
-            num_completions_term *
-            num_ratings_scaler *
-            cls.MULTIPLIER
-        )
-
-        return exploration_impact_score
+        return [exp_models.ExpSummaryModel]
 
     @staticmethod
     def map(item):
         if item.deleted:
             return
 
-        exploration_impact_score = (
-            UserImpactMRJobManager._get_exp_impact_score(item.id))
+        exponent = 2/float(3)
 
-        if exploration_impact_score > 0:
-            # Get exploration summary and contributor ids,
-            # yield for each contributor.
+        # Get average rating and value per user
+        total_rating = 0
+        for ratings_value in item.ratings:
+            total_rating += item.ratings[ratings_value] * int(ratings_value)
+        if sum(item.ratings.itervalues()):
+            average_rating = total_rating / sum(item.ratings.itervalues())
+            value_per_user = average_rating - 2
+        else:
+            value_per_user = 0
```

Comment:
If value_per_user = 0, you could probably just return here, and skip the rest of the calculations?